src/compute/src/render/reduce.rs

```diff
@@ -637,32 +660,43 @@ where
         // Build a series of stages for the reduction
         // Arrange the final result into (key, Row)
         let debug_name = debug_name.to_string();
-        partial.arrange_named::<RowSpine<_,Vec<Row>,_,_>>("Arrange ReduceMinsMaxes")
+        partial
+            .arrange_named::<RowSpine<_, Vec<Row>, _, _>>("Arrange ReduceMinsMaxes")
             .reduce_abelian::<_, RowSpine<_, _, _, _>>("ReduceMinsMaxes", {
-            let mut row_buf = Row::default();
-            move |_key, source, target| {
-                // Negative counts would be surprising, but until we are 100% certain we wont
-                // see them, we should report when we do. We may want to bake even more info
-                // in here in the future.
-                if source.iter().any(|(_val, cnt)| cnt < &0) {
-                    // XXX: This reports user data, which we perhaps should not do!
-                    for (val, cnt) in source.iter() {
-                        if cnt < &0 {
-                            warn!("[customer-data] Negative accumulation in ReduceMinsMaxes: {val:?} with count {cnt:?} in dataflow {debug_name}");
-                            soft_assert_or_log!(false, "Negative accumulation in ReduceMinsMaxes");
+                let mut row_buf = Row::default();
+                move |_key, source, target| {
+                    // Negative counts would be surprising, but until we are 100% certain we wont
+                    // see them, we should report when we do. We may want to bake even more info
+                    // in here in the future.
+                    if validating && source.iter().any(|(_val, cnt)| cnt < &0) {
+                        // XXX: This reports user data, which we perhaps should not do!
+                        for (val, cnt) in source.iter() {
+                            if cnt < &0 {
+                                warn!(
+                                    "[customer-data] Negative accumulation in ReduceMinsMaxes: \
+                                    {val:?} with count {cnt:?} in dataflow {debug_name}"
+                                );
```

Comment:
I think we do not need to go that far, since `warn!` calls in Sentry are put into the breadcrumbs. Only the `error!` calls are relevant to create a top-level issue. And there is some de-duplication there going on based on fingerprinting (which I do not know all the details of).

In any event, right now some invalid accumulation errors are being grouped together even though they have different messages: https://materializeinc.sentry.io/issues/3934284119/events/latest/?project=6780145&query=is%3Aunresolved&referrer=latest-event. Note in the latter that the message in the event differs from the message in the headline, but Sentry assumed the two events should be aggregated. So if anything, we are grouping even more than necessary, from what it seems.

This behavior did not bother me too much, since we will anyway need to be careful to disentangle the error detection ("hey, I saw an invalid accumulation while reducing") from the error cause ("ha, there is some bug in X / customer Y ingested too many retractions / etc"). So we'll probably need to instantiate different issues from the same error messages in Sentry in the future.

Reply:
You are absolutely right! Thanks for clarifying! Regarding Sentry, it seems @philip-stoev manually merged the issues.

Reply:
Apologies :-( Will try to unmerge tomorrow

Reply:
No, I think it's all good! This was just about figuring out how smart Sentry is in merging issues. For posterity, Sentry shows who merged issues under the Activity section.

Reply:
Ah, I see; no worries, and thanks for looking into it!